src/core/data/particles/particle_array.hpp

```diff
@@ -230,6 +230,22 @@ class ParticleArray
     auto& vector() { return particles_; }
     auto& vector() const { return particles_; }
 
+    auto& update_from(This const& that)
+    {
+        bool resize_bigger = this->size() < this->size();
```

Comment:
this this is always false?

Reply:
yep it should be this vs that

Reply:
we might also be able to just delete this function and rely on `operator=` , tbc